finishline

```diff
@@ -78,6 +78,9 @@ def parse_arguments():
                         default='/etc/pki/tls/certs/ca-bundle.crt')
     parser.add_argument('--basic-auth',
                         help='use basic auth instead of Kerberos')
+    parser.add_argument('--auth-path',
+                        help='additional (server) path to authenticate first',
+                        default='/step-auth-gss')
```

Comment:
Defaulting to `/step-auth-gss` would cause users who are using basic auth kerberos without a need to hit another location first to fail. Instead, I'd recommend changing this argument to default to `None`. If the argument is anything other than `None` then modify the `client_kwargs['options']` with `auth_url=args.auth_path`. Otherwise, leave `auth_url` out of the options.

Reply:
Yeah I was worried about that. Will change. Cheers!